src/transactions/TransactionFrame.cpp

```diff
@@ -740,6 +740,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                                   TransactionMeta& outerMeta)
 {
     ZoneScoped;
+    auto& internalErrorCounter = app.getMetrics().NewCounter(
+        {"ledger", "transaction", "internal-error"});
+
+    // We do not want to treat exception as internal-error for non-live ledger
+    // versions. The live ledger version is in the head of the ledger, and
+    // normally should be same as the app.getConfig().LEDGER_PROTOCOL_VERSION,
+    // which should also be same as maxAssertLedgerVersion here. However, right
+    // before a protocol upgrade, the LEDGER_PROTOCOL_VERSION (the software
+    // version) could lead the protocol version by 1, thus we need to offset it
+    // from the maxAssertLedgerVersion.
+    uint32_t maxAssertLedgerVersion =
```

Comment:
We don't do this so it might not be a concern, but it's technically possible for us to bump `CURRENT_LEDGER_PROTOCOL_VERSION` by more than one. If we did this, the internal error would always be downgraded. It might be a good idea to also check that catchup is running (like we do [here](https://github.com/stellar/stellar-core/blob/a44e07d495ec91e55e120557886a6ee5f7363210/src/ledger/LedgerManagerImpl.cpp#L507)) before downgrading the internal error.